src/components/CheckboxGroup/CheckboxGroup.tsx

```diff
@@ -0,0 +1,133 @@
+  onChange?: (names: CheckedNames) => void;
+  variant?: CheckboxGroupVariant;
+}
+
+type ReducerAction =
+  | { type: 'check' | 'uncheck'; name: string }
+  | { type: 'reset'; state: CheckedNames };
+
+const reducer = (state: CheckedNames, action: ReducerAction) => {
+  switch (action.type) {
+    case 'check':
+      return state.concat([action.name]);
+    case 'uncheck':
+      return state.filter((name) => name !== action.name);
+    case 'reset':
+      return action.state;
+  }
+};
+const checkedItems = (items: CheckboxItem[]) =>
+  items.filter(({ checked }) => checked).map(({ name }) => name);
+
+export const CheckboxGroup = ({
+  compact,
+  description,
+  disabled,
+  error,
+  items,
+  legend,
+  onChange,
+  variant = CheckboxGroupVariant.Vertical,
+}: CheckboxGroupProps) => {
+  const allNames = items.map((item) => item.name);
+  if (allNames.length !== new Set(allNames).size) {
+    throw Error('Each name in the checkbox group must be unique.');
+  }
+
+  const randomId = useId();
+
+  const [checkedNames, dispatch] = useReducer(reducer, checkedItems(items));
+
+  useEffect(
+    () => dispatch({ type: 'reset', state: checkedItems(items) }),
+    [items],
+  );
+
+  useEffect(
+    () => (onChange && !disabled ? onChange(checkedNames) : undefined),
+    [checkedNames, onChange, disabled],
+  );
+
+  return (
+    <div
+      className={cn(
+        classes.wrapper,
+        error && classes['wrapper--error'],
+        compact && classes['wrapper--compact'],
+        disabled && classes['wrapper--disabled'],
+      )}
+    >
+      {error && <div className={classes['wrapper--error__line']} />}
+      <fieldset className={classes['checkbox-group']}>
+        {legend && <legend>{legend}</legend>}
+        {description && (
+          <p className={classes['checkbox-group__description']}>
+            {description}
+          </p>
+        )}
+        <div
+          className={cn(
+            classes['checkbox-group__list'],
+            classes[`checkbox-group__list--${variant}`],
+          )}
+        >
+          {items.map((item, index) => (
+            <div key={`checkbox-group-${randomId}-${index}`}>
+              <Checkbox
+                checkboxId={item.id}
+                checked={checkedNames.includes(item.name)}
+                compact={compact}
+                description={item.description}
+                disabled={disabled || item.disabled}
+                error={!!error}
+                label={item.label}
+                name={item.name}
+                onChange={(event) => {
+                  dispatch({
+                    type: event.target.checked ? 'check' : 'uncheck',
+                    name: item.name,
+                  });
+                }}
+              />
+            </div>
+          ))}
+        </div>
+        {error && (
+          <div className={classes['checkbox-group__error-message']}>
+            <ErrorMessage>{error}</ErrorMessage>
+          </div>
+        )}
```

Comment:
This is maybe a bit outside the scope of this issue, but I'm wondering a bit how this will work with checkbox groups when we intend to start using it in the app-frontend code? Isnt the error rendereing completely decoupled from the component itself in that implementation?

Would be good to try and use this component in app-frontend to see how it will work (if you havent already done it 🙂)

Reply:
I don't know the code in app-frontend very well, but I have taken a look and it appears to require some special handling indeed. I see that `CheckboxContainerComponent` takes a `validationMessages` prop that might be used to find a proper input string for `CheckboxGroup`'s `error` prop, but maybe it would be better to add an `errorComponent` prop to `CheckboxGroup`, so that the result of `renderValidationMassegesForComponent()` can be passed to it directly?

Another solution is of course to add an `isValid` prop to `CheckboxGroup` and render it in the same way as the other components in app-frontend. In that case, I think it also would be useful to add a component to wrap the checkboxes and the error message together to apply proper styling, a bit similar to `InputWrapper`, but available for the consumer.

Reply:
Could also perhaps change `error` to accept `React.ReactNode` instead? Then strings and components will be valid.

Not sure I fully understand the other suggestion, lets have a chat about it 🙂